geneselection/datasets/dataset.py

```diff
@@ -2,55 +2,75 @@
 import pandas as pd
 import torch
 from torch.utils.data.dataset import Dataset
+from torch import Tensor
 from collections import Iterable
 from ..utils.dataloader import default_collate
+from typing import Dict, List, Union
 
 
-def gsdataset_from_anndata(adata: anndata.AnnData):
-    return GSDataset(
-        X=torch.from_numpy(adata.X), obs=adata.obs, var=adata.var, uns=adata.uns
-    )
+class GSDatasetVarMismatchError(Exception):
+    """Exception for when asked to merge 2 GSDatasets with different var classes (columns)"""
+    pass
 
 
 class GSDataset(Dataset):
     def __init__(
-        self, X=torch.zeros(1, 1), obs=pd.DataFrame([0]), var=pd.DataFrame([0]), uns={}
+        self,
+        x: Tensor=torch.zeros(1, 1),
+        obs: pd.DataFrame=pd.DataFrame([0]),
+        var: pd.DataFrame=pd.DataFrame([0]),
+        uns: dict={}
     ):
         """
         A data provider class for the larger project. The idea is to capture an AnnData and then
         serve up the rows as either pytorch Tensors, or numpy ndarrays based on the keyword argument
-        :param data: An AnnData object which is copied to protect from changes to the original object.
+        :param X: a Tensor created from for example an (ndarray)AnnData.X data block
+        :param obs: a pandas.DataFrame containing the observational labels corresponding to AnnData.obs
+        :param var: a pandas.DataFrame containing the variable labels contained in an AnnData.obs
+        :param uns: a pandas.DataFrame containing the unstructured descriptors contained in AnnData.uns
```

Comment:
this should be a dict
```suggestion
        :param uns: a dict containing the unstructured descriptors contained in AnnData.uns
```

Reply:
fixed in commit [dac9679](https://github.com/AllenCellModeling/geneselection/commit/dac9679f5fe935554321300b5af50e25cf0a3406)